store/postgres/src/block_range.rs

```diff
@@ -132,6 +132,66 @@ impl<'a> QueryFragment<Pg> for BlockRangeUpperBoundClause<'a> {
     }
 }
 
+/// Helper for generating SQL fragments for selecting entities in a specific block range
+#[derive(Debug, Clone, Copy)]
+pub enum EntityBlockRange {
+    Mutable(BlockRange), // TODO: check if this is a proper type here (maybe Range<BlockNumber>?)
+    Immutable(BlockRange),
+}
+
+impl EntityBlockRange {
+    pub fn new(immutable: bool, block_range: std::ops::Range<BlockNumber>) -> Self {
```

Comment:
IMHO, it would be nicer to pass a `&Table` here so that the caller says 'make a block range for this table' and doesn't need to worry about what that entails